local/getDevices.go

```diff
@@ -192,11 +193,12 @@ func (h *discoveryHandler) Handle(ctx context.Context, d *core.Device) {
 			return
 		}
 		refDev, stored, err = h.deviceCache.TryStoreDeviceToTemporaryCache(newRefDev)
-		if !stored {
+		if err != nil {
 			newRefDev.Release(ctx)
 			return
 		}
-		if err != nil {
+		if !stored {
+			newRefDev.Release(ctx)
 			return
 		}
```

Comment:
I'm not sure I understand this change. There is the same piece of code In both `if`s.

Reply:
It just means that we don't want to use the discovered device in the handler so we release it.